src/main/java/nl/ordina/jobcrawler/service/VacancyStarter.java

```diff
@@ -73,4 +75,20 @@ public void scrape() throws IOException {
         log.info("Finished scraping");
         allVacancies.clear();
     }
+
+    @Scheduled(cron = "0 0 * * * *") // at second 0, minute 0 run this function
+    public void deleteNonExistingVacancies(){
+        log.info("Started deleting non-existing jobs");
+        List<Vacancy> allVacancies = vacancyService.getAllVacancies();
+        List<Vacancy> vacanciesToDelete = allVacancies.stream()
+                .filter(vacancy -> !vacancy.checkURL()) //if the url is not good anymore add it in the vacanciesToDelete
```

Comment:
`vacancy.checkURL()` will not work for Yacht vacancies that do not exist anymore.
An example url for a removed Yacht vacancy is https://www.yacht.nl/vacatures/9079222/data-analist---modeler which returns status 200.

Quick research shows me that if you use a random vacancy url for HuxleyIT it returns status 410. Jobbird returns status 301.

I might fix this myself this week as you should be enjoying your time :-)

Reply:
Resolved https://github.com/mostafa-221/jobcrawler-backend/pull/19/commits/ff26b0d0770003ab93ead9abb80fd9c414483941